test/inject.test.js

```diff
@@ -154,11 +156,10 @@ describe("inject based context", () => {
                 )
             }
         }
-        withConsole(() => {
-            expect(() => renderer.create(<A />)).toThrow(
-                /Store 'foo' is not available! Make sure it is provided by some Provider/
-            )
-        })
+        expect(() => renderer.create(<A />)).toThrow(
```

Comment:
Generally, I try to limit the area where I mock out console in test because it can also swallow other things. Especially painful when trying to console.log something and wondering why it's not there :) That's what `withConsole` was good for. Perhaps we can keep that wrapper but use jest variant inside?

Reply:
What do you think about moving the `mockConsole` invocation right before this expect?

Reply:
I've moved the `mockConsole` invocations as close as I can to where they are needed.

Reply:
Why not keep that `withConsole` as I drafted? It's very handy as you don't need to remember to restore...

```js
function withConsole(cb => {
  const restoreConsole = mockConsole()
  cb()
  restoreConsole()
})
```

Reply:
I'm not sure yet that introducing additional abstraction around the jest-mock-console is worth it, but I'll do it because I'm not 100% sure.

Reply:
It's just for convenience, I don't exactly insist, it just looks safer that way.

Reply:
OK. Thanks. Then it is the final version of PR.